contracts/src/tokens/migration/TokenMigration.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ITokenMigration} from "./ITokenMigration.sol";
+
+// libraries
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+import {CustomRevert} from "contracts/src/utils/libraries/CustomRevert.sol";
+import {TokenMigrationStorage} from "./TokenMigrationStorage.sol";
+import {Validator} from "contracts/src/utils/Validator.sol";
+
+// contracts
+import {PausableBase} from "contracts/src/diamond/facets/pausable/PausableBase.sol";
+import {OwnableBase} from "contracts/src/diamond/facets/ownable/OwnableBase.sol";
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+import {ReentrancyGuard} from "contracts/src/diamond/facets/reentrancy/ReentrancyGuard.sol";
+
+contract TokenMigrationFacet is
+  OwnableBase,
+  PausableBase,
+  ReentrancyGuard,
+  Facet,
+  ITokenMigration
+{
+  using SafeTransferLib for address;
+
+  function __TokenMigrationFacet_init(
+    IERC20 oldToken,
+    IERC20 newToken
+  ) external onlyInitializing {
+    TokenMigrationStorage.Layout storage ds = TokenMigrationStorage.layout();
+    (ds.oldToken, ds.newToken) = (oldToken, newToken);
+    _pause();
+  }
+
+  function migrate(address account) external whenNotPaused nonReentrant {
```

Comment:
Since this method can be called by anyone, does it make sense to check `oldToken.allowance(account, address(this)) == currentBalance)` and reverting early on to save gas on cases where transfer is not approved yet?

Reply:
Yep, fixed in latest push.